### PR TITLE
Add Trino 454 release notes

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 454 release.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 26 July 2024

* #22798 ✅ rn ✅ docs
* #22785 ✅ rn ✅ docs
* #22355 ✅ rn ✅ docs
* #22823 ✅ rn ✅ docs
* #22826 ✅ rn ✅ docs
* #22830 ✅ rn ✅ docs
* #22837 ✅ rn ✅ docs
* #22838 ✅ rn ✅ docs
* #22835 ✅ rn ✅ docs
* #22828 ✅ rn ✅ docs
* #22825 ✅ rn ✅ docs
* #22843 ✅ rn ✅ docs
* #22794 ✅ rn ✅ docs
* #22844 ✅ rn ✅ docs

## 28 July 2024

* #22821 ✅ rn ✅ docs

## 29 July 2024

* #22834 ✅ rn ✅ docs
* #22783 ✅ rn ✅ docs
* #20843 ✅ rn ✅ docs
* #22852 ✅ rn ✅ docs
* #22855 ✅ rn ✅ docs
* #22856 ✅ rn ✅ docs
* #22829 ✅ rn ✅ docs
* #22862 ✅ rn ✅ docs
* #22261 ✅ rn ✅ docs
* #22866 ✅ rn ✅ docs
* #22836 ✅ rn ✅ docs
* #22820 ✅ rn ✅ docs
* #22839 ✅ rn ✅ docs
* #22854 ✅ rn ✅ docs

## 30 July 2024

* #22857 ✅ rn ✅ docs
* #22706 ✅ rn ✅ docs
* #22864 ✅ rn ✅ docs
* #22876 ✅ rn ✅ docs
* #22878 ✅ rn ✅ docs
* #22841 ❌ rn ✅ docs
* #22880 ✅ rn ✅ docs
* #22559 ✅ rn ✅ docs
* #22858 ✅ rn ✅ docs
* #22863 ✅ rn ✅ docs
* #22883 ✅ rn ✅ docs

## 31 July 2024

* #22884 ✅ rn ✅ docs
* #22879 ✅ rn ✅ docs
* #22881 ✅ rn ✅ docs
* #22882 ✅ rn ✅ docs
* #22892 ✅ rn ✅ docs
* #22904 ✅ rn ✅ docs
* #22903 ✅ rn ✅ docs

## 1 Aug 2024

* #22739 ✅ rn ✅ docs
* #22584 ✅ rn ✅ docs
* #22891 ✅ rn ✅ docs
* #22910 ✅ rn ✅ docs
* #22895 ✅ rn ✅ docs

## 2 Aug 2024

* #22577 ✅ rn ✅ docs
* #22909 ✅ rn ✅ docs
* #22918 ✅ rn ✅ docs
* #22900 ✅ rn ✅ docs
* #22925 ✅ rn ✅ docs

## 3 Aug 2024

* #22893 ✅ rn ✅ docs

## 4 Aug 2024

* #21463 ✅ rn ✅ docs
* #22850 ✅ rn ✅ docs
* #22929 ✅ rn ✅ docs

## 5 Aug 2024

* #22911 ✅ rn ✅ docs
* #22931 ✅ rn ✅ docs
* #22939 ✅ rn ✅ docs
* #22940 ✅ rn ✅ docs
* #22924 ✅ rn ✅ docs
* #22942 ✅ rn ✅ docs
* #22938 ✅ rn ✅ docs

## 6 Aug 2024

* #22950 ✅ rn ✅ docs
* #22948 ✅ rn ✅ docs
* #22952 ✅ rn ✅ docs
* #22933 ✅ rn ✅ docs
* #22964 ✅ rn ✅ docs

## 7 Aug 2024

* #22959 ✅ rn ✅ docs
* #22966 ✅ rn ✅ docs
* #22958 ✅ rn ✅ docs
* #22670 ✅ rn ❌ docs
* #22969 ✅ rn ✅ docs
* #22633 ✅ rn ✅ docs

## 8 Aug 2024

* #22618 ✅ rn ✅ docs
* #22980 ✅ rn ✅ docs
* #22981 ✅ rn ✅ docs
* #22953 ✅ rn ✅ docs
* #22979 ✅ rn ✅ docs

## 9 Aug 2024

* #22988 ✅ rn ✅ docs
* #22985 ✅ rn ✅ docs
* #22990 ✅ rn ✅ docs
* #22991 ✅ rn ✅ docs
* #22971 ✅ rn ✅ docs
* #22945 ✅ rn ✅ docs
* #22963 ✅ rn ✅ docs

## 10 Aug 2024

* #22999 ✅ rn ✅ docs

## 12 Aug 2024

* #23001 ✅ rn ✅ docs
* #23006 ✅ rn ✅ docs
* #23010 ✅ rn ✅ docs
* #22814 ✅ rn ✅ docs

## 13 Aug 2024

* #22936 ✅ rn ✅ docs
* #23015 ✅ rn ✅ docs
* #23012 ✅ rn ✅ docs
* #23020 ✅ rn ✅ docs
* #23016 ✅ rn ✅ docs
* #23017 ✅ rn ✅ docs
* #22869 ✅ rn ✅ docs
* #23023 ✅ rn ✅ docs
* #23022 ✅ rn ✅ docs
* #22888 ✅ rn ❌  docs
* #23031 ✅ rn ✅ docs

## 14 Aug 2024

* #23033 ✅ rn ✅ docs
* #22102 ✅ rn ✅ docs
* #23036 ✅ rn ✅ docs
* #23007 ✅ rn ✅ docs
* #22987 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-454.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Release 454 (7 Aug 2024)
+
+## General
+
+* Add support for retrieving lightweight query information from the `QueryInfo`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:5]: There's no such a thing as "QueryInfo" endpoint. The endpoint is `/v1/query`

File: docs/src/main/sphinx/release/release-454.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  endpoint with the new `pruned` property. ({issue}`20843`)
+* Improve performance of queries with multiple distinct aggregations. ({issue}`22355`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:7]: "... with multiple aggregations involving `DISTINCT`"

File: docs/src/main/sphinx/release/release-454.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+* Skip query completed notifications in the event listener when there are too
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:8]: This should describe the user visible impact and the reason why this matters. It should be something like: "Improve server stability by discarding events when event listeners are slow to process them.

BTW, we'll probably have to change the name of the property, as "concurrent" doesn't convey the right notion. See my comment: https://github.com/trinodb/trino/pull/22911#discussion_r1708076899

cc @sopel39

File: docs/src/main/sphinx/release/release-454.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  many concurrent completion requests. The maximum number of requests to handle
+  before skipping can be configured with the
+  `event-listener.max-concurrent-query-completed-events` configuration property. ({issue}`22911`)
+* Fix failure for queries involving {func}`json_parse` and a cast to `array`,
+  `map`, or `row`. ({issue}`22840`)
+
+## Security
+
+* Fix the Trino user incorrectly defaulting to the `user.name` property when no
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:17]: Is this for the JDBC driver? CLI? It should be moved to the appropriate section(s).

Also, the phrase "`user.name` property" doesn't mean anything for a user of the CLI or JDBC driver. Is it the user running the process? cc @wendigo

File: docs/src/main/sphinx/release/release-454.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  `Connection.prepareStatement` method. ({issue}`22100`)
+
+## BigQuery connector
+
+* Add support for the `execute` procedure. ({issue}`22692`)
+
+## Delta Lake connector
+
+* Add S3 security mapping for the native S3 filesystem. ({issue}`22559`)
+* Add support for caching table metadata to the metastore. ({issue}`21463`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:37]: Not sure what this means. What's the user-visible impact? cc @ebyhr

File: docs/src/main/sphinx/release/release-454.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+## Hudi connector
+
+* Add S3 security mapping for the native S3 filesystem. ({issue}`22559`)
+* {{breaking}} Rename metastore OpenTelemetry attributes from `trino.hive.*` to
+  `trino.metastore.*`. ({issue}`22866`)
+
+## Iceberg connector
+
+* Add support for the [Polaris catalog](https://polaris.io/). ({issue}`22909`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:62]: That PR doesn't add support for Polaris. It just adds some tests for existing behavior. Remove this item.

**Review by martint — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-454.md (hunk 5 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-454.md:37]: This improves metadata listing speed from user perspective.

File: docs/src/main/sphinx/release/release-454.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
+# Release 454 (7 Aug 2024)
+
+## General
+
+* Add support for retrieving lightweight query information from the `/v1/query`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:5]: "with the `pruned` query parameter"

File: docs/src/main/sphinx/release/release-454.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  endpoint with the new `pruned` property. ({issue}`20843`)
+* Improve performance of queries with multiple aggregations involving
+  `DISTINCT`. ({issue}`22355`)
+* Improve server stability by discarding query completed events when event
```
Comment (martint) [inline docs/src/main/sphinx/release/release-454.md:9]: "The maximum number pending events before new events are discarded can be configured with the ..."

File: docs/src/main/sphinx/release/release-454.md (hunk 4 above)
Comment (wendigo) [reply docs/src/main/sphinx/release/release-454.md]: yes, that's the user running the process and this is for JDBC driver

**Review by martint — APPROVED** [review]

**Pull request merged**